fastlane/lib/fastlane/actions/get_version_number.rb

```diff
@@ -26,6 +26,16 @@ def self.run(params)
           version_number = get_version_number_from_build_settings!(target, $1, configuration)
         end
 
+        # Check if version_number is not set and try retrieving value from project settings
+        if version_number.nil?
+          version_number = get_version_number_from_build_settings!(project, $1, configuration)
```

Comment:
This usage of $1 here seems to be a bit confusing 🤔 It isn't really tied to any specific regex here. I think this fallback almost needs to go inside the if statements above. Thoughts?

Reply:
you are absolutely correct, the downside would be having the same line twice, one in each if check